contracts/truefi2/TrueCreditAgency.sol

```diff
@@ -361,23 +364,62 @@ contract TrueCreditAgency is UpgradeableClaimable {
 
     function poke(ITrueFiPool2 pool) public {
         uint256 bitMap = usedBucketsBitmap;
-        CreditScoreBucket[256] storage creditScoreBuckets = buckets[pool];
         uint256 timeNow = block.timestamp;
         uint256 partialRate = riskPremium.add(utilizationAdjustmentRate(pool));
 
+        for (uint16 i = 0; i <= MAX_CREDIT_SCORE; (i++, bitMap >>= 1)) {
+            if (bitMap & 1 == 0) {
+                continue;
+            }
+
+            _pokeSingleBucket(pool, uint8(i), timeNow, partialRate);
+        }
+    }
+
+    function pokeSingleBucket(ITrueFiPool2 pool, uint8 bucketNumber) internal {
+        uint256 timeNow = block.timestamp;
+        uint256 partialRate = riskPremium.add(utilizationAdjustmentRate(pool));
+
+        _pokeSingleBucket(pool, bucketNumber, timeNow, partialRate);
+    }
+
+    function _pokeSingleBucket(
+        ITrueFiPool2 pool,
+        uint8 bucketNumber,
+        uint256 timeNow,
+        uint256 partialRate
+    ) internal {
+        CreditScoreBucket storage bucket = buckets[pool][bucketNumber];
+
+        poolTotalInterest[pool] = poolTotalInterest[pool].add(
+            bucket.rate.mul(1e23).mul(bucket.totalBorrowed).mul(timeNow.sub(bucket.timestamp)).div(365 days)
```

Comment:
My previous comment was closed but not addressed: https://github.com/trusttoken/smart-contracts/pull/797#discussion_r681832229.

Let's extract the `bucket.rate.mul(1e23).mul(timeNow.sub(bucket.timestamp)).div(365 days)` partial calculation (that gets duplicated throughout this contract) into a function.